experimental/adaptive-tool/server/lg/src/providers/completion.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	CompletionItem, CompletionItemKind, TextDocumentPositionParams,	Files, TextDocuments
+} from 'vscode-languageserver';
+
+import {
+	TextDocument, Position
+} from 'vscode-languageserver-textdocument';
+import * as util from '../util';
+import { TemplatesStatus } from '../templatesStatus';
+import * as path from 'path';
+
+export function provideCompletionItems(_textDocumentPosition: TextDocumentPositionParams, documents: TextDocuments<TextDocument>){
+	const document = documents.get(_textDocumentPosition.textDocument.uri);
+	const fspath = Files.uriToFilePath(document!.uri);
+	const position = _textDocumentPosition.position;
+	const lineTextBefore = document?.getText({
```

Comment:
As I know, [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) first appeared on ts 3.7 version. So, you should make sure all the compile tools (dependency, typescript version of vscode) should be newer than version 3.7.

Reply:
fixed